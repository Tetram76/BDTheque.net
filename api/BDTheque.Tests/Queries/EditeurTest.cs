namespace BDTheque.Tests.Queries;

using Xunit.Abstractions;

public class EditeurTest(ITestOutputHelper testOutputHelper) : DatabaseTest(testOutputHelper)
{
    [Fact]
    public async Task GetEditeursTest()
    {
        Editeur[] editeurs =
        [
            new Editeur
            {
                Nom = "Aa"
            },
            new Editeur
            {
                Nom = "Bb"
            },
            new Editeur
            {
                Nom = "Cc"
            }
        ];

        RegisterFinalizer(() => RemoveEntities(editeurs));
        await AddEntities(editeurs);

        string response = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(
                """
                query {
                    editeurList(last: 10, where: {initiale: {eq: "A"}}, order: {nom: ASC}) {
                      pageInfo {
                        hasNextPage
                        hasPreviousPage
                      }
                      edges {
                        cursor
                        node {
                          id
                          nom
                        }
                      }
                    }
                }
                """
            )
        );

        response.Should().MatchSnapshot(options => options.IgnoreAllFields("id"));
    }
}
