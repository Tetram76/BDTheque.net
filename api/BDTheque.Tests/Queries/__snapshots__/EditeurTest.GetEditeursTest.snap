{
  "data": {
    "editeurList": {
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false
      },
      "edges": [
        {
          "cursor": "MA==",
          "node": {
            "id": "d13d3f5dcb434a6ab187cff4c648339b",
            "nom": "Aa"
          }
        }
      ]
    }
  }
}
